### PR TITLE
fix: WDIO Cucumber Test Tags

### PR DESCRIPTION
**Pull request by sauravdas1997**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

Currently Browserstack accessibility is not running when tags are passed in accessibilityOptions in case of cucumber, so we are handling the use case for test tags in cucumber.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by erwinheitzman** [comment]
Hi @sauravdas1997 thank you for your contribution!

I have some requests:
1. file an issue that can be coupled to this PR so that we can first verify and discuss if this is an issue
2. describe the changes made
3. add some tests to verify the changed behaviour

Thank you

**Comment by erwinheitzman** [comment]
Hi @sauravdas1997 thank you for your contribution!

This does indeed need to be `name`, however the value also shouldn't be an array, could you make this change and update/add tests were needed? Thank you.

File: packages/wdio-browserstack-service/tests/util.test.ts [hunk 1]
```diff
@@ -1026,13 +1026,30 @@ describe('validateCapsWithA11y', () => {
...
+                }
+            ]
+        }
+    }
     it('returns true if full test name contains includeTags', async () => {
         expect(shouldScanTestForAccessibility('suite title', 'test title', { includeTagsInTestingScope: 'title' })).toEqual(true)
     })
 
-    it('returns false if full test name contains excludeTags', async () => {
+    it.only('returns false if full test name contains excludeTags', async () => {
```
Comment (nextlevelbeard) [inline packages/wdio-browserstack-service/tests/util.test.ts:1042]: Don't forget to remove this `.only`

Comment (sauravdas1997) [reply packages/wdio-browserstack-service/tests/util.test.ts]: done

**Comment by sauravdas1997** [comment]
> file an issue that can be coupled to this PR so that we can first verify and discuss if this is an issue
This is a fix for a browserstack specific offering that we came across during our internal testing.
> describe the changes made
Added in the PR description
> add some tests to verify the changed behaviour
Added in PR

File: packages/wdio-browserstack-service/src/util.ts [hunk 2]
```diff
@@ -343,11 +343,20 @@ export const validateCapsWithA11y = (deviceName?: any, platformMeta?: { [key: st
...
+export const shouldScanTestForAccessibility = (suiteTitle: string | undefined, testTitle: string, accessibilityOptions?: { [key: string]: any; }, world?: { [key: string]: any; }, isCucumber?: boolean ) => {
     try {
         const includeTags = Array.isArray(accessibilityOptions?.includeTagsInTestingScope) ? accessibilityOptions?.includeTagsInTestingScope : []
         const excludeTags = Array.isArray(accessibilityOptions?.excludeTagsInTestingScope) ? accessibilityOptions?.excludeTagsInTestingScope : []
 
+        if (isCucumber) {
+            const tagsList: string[] = []
+            world?.pickle?.tags.map((tag: { [key: string]: any; }) => tagsList.push(tag.name))
+            const excluded = excludeTags?.some((exclude) => tagsList.includes(exclude))
+            const included = includeTags?.length === 0 || includeTags?.some((include) => tagsList.includes(include))
```
Comment (erwinheitzman) [inline packages/wdio-browserstack-service/src/util.ts:355]: Could you explain why we do an explicit check here `includeTags?.length === 0`? I feel like the include and exclude arrays should be treated the same way so it's a surprise to see this.

**Review by erwinheitzman — COMMENTED** [review]
Got one question as a comment, looks got overall!

File: packages/wdio-browserstack-service/src/util.ts (hunk 2 above)
Comment (sauravdas1997) [reply packages/wdio-browserstack-service/src/util.ts:355]: Use case: If the user has not specified any includeTags, we should return the shouldScanTestForAccessibility results as true. However, if the user has mentioned a value in includeTags, we should evaluate against it. If the user does not provide excludeTags, no tags should be excluded by default. We will only honour it if the user explicitly specifies tags in excludeTags.

**Review by christian-bromann — APPROVED** [review]
LGTM 👍 

@erwinheitzman any objections to merge?

**Comment by christian-bromann** [comment]
I will go ahead and merge. Happy to address remaining issues in a new PR.

**Pull request merged**